tests/unit/validator_tests/test_validate.py

```diff
@@ -229,6 +230,121 @@ def test_valid_wo_opt_attr(self):
+        child_groups = [GroupBuilder(f'bar_{n}', attributes={'data_type': 'Bar'}) for n in range(n_groups)]
+        child_datasets = [DatasetBuilder(f'baz_{n}', n, attributes={'data_type': 'Baz'}) for n in range(n_datasets)]
+        child_links = [LinkBuilder(GroupBuilder(f'qux_{n}', attributes={'data_type': 'Qux'}), f'qux_{n}_link')
+                       for n in range(n_links)]
+        return GroupBuilder('my_foo', attributes={'data_type': 'Foo'},
+                            groups=child_groups, datasets=child_datasets, links=child_links)
+
+    def test_valid_zero_or_many(self):
+        """"Verify that groups/datasets/links with ZERO_OR_MANY and a valid quantity correctly pass validation"""
+        specs = self.create_test_specs(q_groups=ZERO_OR_MANY, q_datasets=ZERO_OR_MANY, q_links=ZERO_OR_MANY)
+        self.configure_specs(specs)
+        for n in [0, 1, 2, 5]:
+            with self.subTest(quantity=n):
+                builder = self.get_test_builder(n_groups=n, n_datasets=n, n_links=n)
+                results = self.vmap.validate(builder)
+                self.assertEqual(len(results), 0)
+
+    def test_valid_one_or_many(self):
+        """"Verify that groups/datasets/links with ONE_OR_MANY and a valid quantity correctly pass validation"""
+        specs = self.create_test_specs(q_groups=ONE_OR_MANY, q_datasets=ONE_OR_MANY, q_links=ONE_OR_MANY)
+        self.configure_specs(specs)
+        for n in [1, 2, 5]:
+            with self.subTest(quantity=n):
+                builder = self.get_test_builder(n_groups=n, n_datasets=n, n_links=n)
+                results = self.vmap.validate(builder)
+                self.assertEqual(len(results), 0)
+
+    def test_valid_zero_or_one(self):
+        """"Verify that groups/datasets/links with ZERO_OR_ONE and a valid quantity correctly pass validation"""
+        specs = self.create_test_specs(q_groups=ZERO_OR_ONE, q_datasets=ZERO_OR_ONE, q_links=ZERO_OR_ONE)
+        self.configure_specs(specs)
+        for n in [0, 1]:
+            with self.subTest(quantity=n):
+                builder = self.get_test_builder(n_groups=n, n_datasets=n, n_links=n)
+                results = self.vmap.validate(builder)
+                self.assertEqual(len(results), 0)
+
+    def test_valid_fixed_quantity(self):
+        """"Verify that groups/datasets/links with a correct fixed quantity correctly pass validation"""
+        self.configure_specs(self.create_test_specs(q_groups=2, q_datasets=3, q_links=5))
+        builder = self.get_test_builder(n_groups=2, n_datasets=3, n_links=5)
+        results = self.vmap.validate(builder)
+        self.assertEqual(len(results), 0)
+
+    def test_missing_one_or_many_should_not_return_incorrect_quantity_error(self):
+        """Verify that missing ONE_OR_MANY groups/datasets/links should not return an IncorrectQuantityError"""
+        specs = self.create_test_specs(q_groups=ONE_OR_MANY, q_datasets=ONE_OR_MANY, q_links=ONE_OR_MANY)
```

Comment:
```suggestion
        # NOTE: a MissingDataType error should be returned instead
        specs = self.create_test_specs(q_groups=ONE_OR_MANY, q_datasets=ONE_OR_MANY, q_links=ONE_OR_MANY)
```

Reply:
Good suggestion. I put it in the docstring for the test function as it seems to me like it makes sense to be part of the test documentation.